cognite_toolkit/_cdf_tk/loaders/_resource_loaders/function_loaders.py

```diff
@@ -533,22 +523,33 @@ def retrieve(self, ids: SequenceNotStr[FunctionScheduleID]) -> FunctionSchedules
 
     def create(self, items: FunctionScheduleWriteList) -> FunctionSchedulesList:
         created_list = FunctionSchedulesList([], cognite_client=self.client)
+        functions_to_lookup = list({item.function_external_id for item in items if item.function_external_id})
+        function_id_by_external_id: dict[str, int] = {}
+        if functions_to_lookup:
+            function_ids = self.client.lookup.functions.id(functions_to_lookup)
+            function_id_by_external_id = dict(zip(functions_to_lookup, function_ids))
```

Comment:
This is avoids the N+1 problem. Now all function IDs are looked up in one request, instead of one request per schedule.